Error: invalid conversion from 'AVCodecContext* const*' to 'AVCodecContext**' [-fpermissive]

### Platform

libretro / Retroarch

### Compiler and build tool versions

CMAKE 3.24.2

### Operating system version

ubuntu 22.04 toolchain Emuelec Odroid n2+

### Build commands used

PROJECT=Amlogic-ce DEVICE=Amlogic-ng ARCH=aarch64 DISTRO=EmuELEC ./scripts/build ppsspp
PROJECT=Amlogic-ce DEVICE=Amlogic-ng ARCH=aarch64 DISTRO=EmuELEC ./scripts/build PPSSPPSDL

When compiling PPSSPPSDL retroarch and regular version it is failing compiling with following error starting on commit https://github.com/hrydgard/ppsspp/commit/dbe6ec80a0694d2c990697f7676fbe197595e27f

Error issue Error: invalid conversion from 'AVCodecContext* const*' to 'AVCodecContext**' [-fpermissive]

I'm using latest commit https://github.com/hrydgard/ppsspp/commit/777eae7398c7af293a7cb0aa01a95cf5eb9905dc

Error Log


[error ppsspp.txt](https://github.com/user-attachments/files/19997106/error.ppsspp.txt)


### What happens

/build/ppsspp-777eae7398c7af293a7cb0aa01a95cf5eb9905dc/Core/HW/MediaEngine.cpp:361:38: error: invalid conversion from 'AVCodecContext* const*' to 'AVCodecContext**' [-fpermissive]

### PPSSPP version affected

777eae7398c7af293a7cb0aa01a95cf5eb9905dc

### Last working version

62e4b2852c1313435a44f82f02c535ec85e49a4c

### Checklist

- [x] Test in the [latest git build](https://www.ppsspp.org/devbuilds) in case it's already fixed.
- [x] Make sure to run `git submodule update --init --recursive` before building.
- [x] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [x] Try deleting the build directory and running the build again.
- [x] Check GitHub Actions, which builds every merge and PR.
- [x] Include logs and help us reproduce.

--- a/Core/HW/MediaEngine.cpp
+++ b/Core/HW/MediaEngine.cpp
@@ -343,8 +343,7 @@ bool MediaEngine::openContext(bool keepReadPos) {
 	return true;
 }
 
-void MediaEngine::closeContext()
-{
+void MediaEngine::closeContext() {
 #ifdef USE_FFMPEG
 	if (m_buffer)
 		av_free(m_buffer);
@@ -356,7 +355,7 @@ void MediaEngine::closeContext()
 		av_free(m_pIOContext->buffer);
 	if (m_pIOContext)
 		av_free(m_pIOContext);
-	for (const auto &it : m_pCodecCtxs) {
+	for (auto &it : m_pCodecCtxs) {
 #if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(57, 33, 100)
 		avcodec_free_context(&it.second);
 #else
@@ -365,7 +364,7 @@ void MediaEngine::closeContext()
 	}
 	m_pCodecCtxs.clear();
 	// These are streams allocated from avformat_new_stream.
-	for (const auto &it : m_codecsToClose) {
+	for (auto &it : m_codecsToClose) {
 		avcodec_close(it);
 	}
 	m_codecsToClose.clear();

--- a/Core/HW/MediaEngine.h
+++ b/Core/HW/MediaEngine.h
@@ -113,7 +113,7 @@ public:  // TODO: Very little of this below should be public.
 
 	int m_desWidth = 0;
 	int m_desHeight = 0;
-	int m_bufSize;
+	int m_bufSize;  // initialized in constructor
 	s64 m_videopts = 0;
 
 	s64 m_firstTimeStamp = 0;

